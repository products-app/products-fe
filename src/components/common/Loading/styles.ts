import { styled, keyframes, Text } from '@lebernardo/react'

export const LoadingContainer = styled('div', {
  display: 'flex',
  justifyItems: 'center',
  alignItems: 'center',
  flexDirection: 'column',
  padding: '8rem 0',
})

const spinAnimation = keyframes({
  '0%': { transform: 'rotate(0deg)' },
  '100%': { transform: 'rotate(360deg)' },
})

export const LoadingIcon = styled('div', {
  borderColor: 'rgb(209 213 219)',
  width: '2.5rem',
  height: '2.5rem',
  animation: `${spinAnimation} 1s linear infinite`,
  borderRadius: '9999px',
  borderWidth: '8px',
  borderTopColor: '$base500',
  marginBottom: '2rem',
})

export const TextContainer = styled(Text, {
  color: '$gray500',
})
