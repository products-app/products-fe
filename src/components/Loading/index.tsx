import { LoadingContainer, LoadingIcon, TextContainer } from './styles'

type LoadingProps = {
  message?: string
}

const Loading = ({ message }: LoadingProps) => {
  return (
    <LoadingContainer>
      <LoadingIcon />
      <TextContainer size="lg">{message}</TextContainer>
    </LoadingContainer>
  )
}

const defaultProps = {
  message: 'Carregando, aguarde...',
}

Loading.defaultProps = defaultProps
export default Loading
