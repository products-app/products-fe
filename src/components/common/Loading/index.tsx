import { css } from '@lebernardo/react'
import { LoadingContainer, LoadingIcon, TextContainer } from './styles'

type LoadingProps = {
  message?: string
  css?: typeof css
}

const Loading = ({ message, css }: LoadingProps) => {
  return (
    <LoadingContainer css={{ css }}>
      <LoadingIcon />
      {message && <TextContainer size="lg">{message}</TextContainer>}
    </LoadingContainer>
  )
}

const defaultProps = {
  message: undefined,
  css: '',
}

Loading.defaultProps = defaultProps
export default Loading
