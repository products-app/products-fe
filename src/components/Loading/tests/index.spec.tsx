import Loading from '../index'
import { render, screen } from '@testing-library/react'

describe('Loading component', () => {
  it('should render correctly', () => {
    const { container } = render(<Loading />)
    expect(container).toMatchSnapshot()
  })

  it('should contain the loading default message', () => {
    render(<Loading />)
    expect(screen.getByText('Carregando, aguarde...')).toBeInTheDocument()
    expect(() => screen.getByText('no exist')).toThrow()
  })

  it('should contain the loading default message', () => {
    render(<Loading message="Test" />)
    expect(screen.getByText('Test')).toBeInTheDocument()
    expect(() => screen.getByText('no exist')).toThrow()
  })
})
