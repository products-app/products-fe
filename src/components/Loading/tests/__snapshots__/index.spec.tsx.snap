// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render correctly 1`] = `
<div>
  <div
    class="c-jjpTuW"
  >
    <div
      class="c-hIpGLj"
    />
    <p
      class="c-dgMvMW c-igsGei c-dgMvMW-eQRsLC-size-lg"
    >
      Carregando, aguarde...
    </p>
  </div>
</div>
`;
